src/main/java/org/folio/entlinks/controller/delegate/AuthoritySourceFileServiceDelegate.java

```diff
@@ -118,4 +122,27 @@ private void validateModifyPossibility(DomainEventType eventType, AuthoritySourc
         List.of(new Parameter("id").value(String.valueOf(entity.getId()))));
     }
   }
+
+  private void validatePatchRequest(AuthoritySourceFilePatchDto patchDto, AuthoritySourceFile existing) {
+    var errorParameters = new LinkedList<Parameter>();
+    var hasAuthorityReferences = service.authoritiesExistForSourceFile(existing.getId());
+
+    if (!(existing.getSource().equals(FOLIO) || hasAuthorityReferences)) {
+      return;
+    }
+
+    if (patchDto.getCodes() != null) {
+      errorParameters.add(new Parameter("codes")
+          .value(String.join(",", patchDto.getCodes())));
+    }
+    if (patchDto.getHridManagement() != null) {
+      errorParameters.add(new Parameter("hridManagement.startNumber")
+          .value(patchDto.getHridManagement().getStartNumber().toString()));
+    }
+
+    if (!errorParameters.isEmpty()) {
+      throw new RequestBodyValidationException(
+          "Unable to patch. Authority source file source is FOLIO or it has authority references", errorParameters);
```

Comment:
I think there are an options to update folio records.

Reply:
And there is also possibility to update some fields when references exist

Reply:
Yes, but this error is thrown only if at least one (FOLIO source / authority exist) condition met and at least one (codes/hrid management) parameter passed. Each of these 2 parameters cannot be updated when any of 2 conditions met